Improve awk script used to filter tagged cheats

lucaslugao opened the pull request:
Most of the code is explained in the comments 😄  Feel free to remove them as well as the argument checking in the awk script 
 -- since it is only used by the script itself.

This PR also:
- Fix variable passing (using `-v` is mandatory in awk)
- Enable [bash strict mode](https://gist.github.com/robin-a-meade/58d60124b88b60816e8349d1e3938615) to prevent future silent errors

giggio commented:
I moved the logic to the .awk file and added tests, can you update so we can review if the tests pass?

lucaslugao commented:
All tests passing:
```
$ bats tests/build.spec.sh 
build.spec.sh
 ✓ Without shell and os
 ✓ With shell, without os
 ✓ Without shell, with os
 ✓ With shell and os
 ✓ With shell, without os, twice
 ✓ Create files

6 tests, 0 failures
```

lucaslugao commented:
Fixed the `--verbose` flag and other minor problems. Also changed the `--files` flag to `--file` following the abseil `DEFINE_multi_string` pattern.

Turned the command substittution running the awk script into a pipeline to prevent error swallowing.

giggio commented:
@lucaslugao build is failing.
What do you think of splitting the PR in two? One for the bash part, another for the awk part?

lucaslugao commented:
I fixed various bugs and improved tests. Also the build is working now :)
